doc/examples/python_example_with_modes.py

```diff
@@ -49,8 +49,8 @@
 for mode in avail_modes:
   for diff in avail_diff:
 
-    ale.setMode(mode)
     ale.setDifficulty(diff)
+    ale.setMode(mode)
```

Comment:
Is there a reason for the change?

Reply:
Yes. I created an issue about that. In Space Invaders, if we set the mode first, when we set the difficulty we end up resetting the mode that was set with softReset. We have to come up with a fix for that to ensure we have a more robust way of setting modes and difficulties. It seems to me this happens only in Space Invaders.